src/authService.js

```diff
@@ -95,21 +88,44 @@ export class AuthService {
   * @returns {Boolean} true: for Non-JWT tokens and unexpired JWT tokens, false: else
   *
   */
-  isAuthenticated() {
-    const isExpired = this.authentication.isTokenExpired();
-    if (isExpired && this.config.autoUpdateToken) {
-      if (this.isRefreshing) {
-        return true;
-      }
-      this.updateToken();
+  isAuthenticated(asPromise) {
+    let authenticated = this.authentication.isAuthenticated();
+
+    // auto-update token?
+    if (!authenticated
+      && this.config.autoUpdateToken
+      && this.authentication.getAccessToken()
+      && this.authentication.getRefreshToken()) {
+      authenticated = this.updateToken();
+    }
+
+    // return as boolean or Promise
+    if (asPromise) {
+      if (authenticated instanceof Promise) return authenticated;
+      return Promise.resolve(authenticated);
     }
-    return this.authentication.isAuthenticated();
+
+    if (authenticated instanceof Promise) {
+      authenticated.catch(()=>{}).then(Promise.resolve);
+      return true;
+    }
+    return authenticated;
+  }
+
+  /**
+   * Gets remaining time in seconds
+   *
+   * @returns {Number} remaing time for JWT tokens, NaN for all other tokesn
+   *
+   */
+  getTimeLeft() {
```

Comment:
I think we should rename this method. timeLeft sounds very ambiguous.  Perhaps `getTtl()`? ttl is a known term.